main.go

```diff
@@ -1,117 +1,142 @@
-		{
-			c := service.Config{
-				Logger: logger,
-
-				Flag:  f,
-				Viper: v,
-			}
-
-			newService, err = service.New(c)
-			if err != nil {
-				panic(microerror.JSON(err))
-			}
-
-			go newService.Boot(ctx)
-		}
-
-		// Create a new custom server which bundles our endpoints.
-		var newServer microserver.Server
-		{
-			c := server.Config{
-				Logger:  logger,
-				Service: newService,
-
-				Viper: v,
-			}
-
-			newServer, err = server.New(c)
-			if err != nil {
-				panic(microerror.JSON(err))
-			}
-		}
-
-		return newServer
-	}
+	ctrl.SetLogger(klogr.New())
 
-	// Create a new microkit command which manages our custom microservice.
-	var newCommand command.Command
-	{
-		c := command.Config{
-			Logger:        logger,
-			ServerFactory: serverFactory,
-
-			Description: project.Description(),
-			GitCommit:   project.GitSHA(),
-			Name:        project.Name(),
-			Source:      project.Source(),
-			Version:     project.Version(),
-		}
-
-		newCommand, err = command.New(c)
-		if err != nil {
-			return microerror.Mask(err)
-		}
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme:                 scheme,
+		MetricsBindAddress:     metricsAddr,
+		Port:                   9443,
+		HealthProbeBindAddress: probeAddr,
+		LeaderElection:         enableLeaderElection,
+		LeaderElectionID:       "a0acdc7d.infrastructure.cluster.x-k8s.io",
+	})
+	if err != nil {
+		setupLog.Error(err, "unable to start manager")
+		os.Exit(1)
 	}
 
-	daemonCommand := newCommand.DaemonCommand().CobraCommand()
+	// Initialize event recorder.
+	record.InitFromRecorder(mgr.GetEventRecorderFor("kvm-controller"))
+
+	if err = (&controllers.KVMClusterReconciler{
+		Client:   mgr.GetClient(),
+		Recorder: mgr.GetEventRecorderFor("kvmcluster-controller"),
+		Scheme:   mgr.GetScheme(),
+	}).SetupWithManager(mgr); err != nil {
+		setupLog.Error(err, "unable to create controller", "controller", "KVMCluster")
+		os.Exit(1)
+	}
+	//+kubebuilder:scaffold:builder
 
-	daemonCommand.PersistentFlags().String(f.Service.Kubernetes.Address, "http://127.0.0.1:6443", "Address used to connect to Kubernetes. When empty in-cluster config is created.")
-	daemonCommand.PersistentFlags().Bool(f.Service.Kubernetes.InCluster, false, "Whether to use the in-cluster config to authenticate with Kubernetes.")
-	daemonCommand.PersistentFlags().String(f.Service.Kubernetes.KubeConfig, "", "KubeConfig used to connect to Kubernetes. When empty other settings are used.")
-	daemonCommand.PersistentFlags().String(f.Service.Kubernetes.TLS.CAFile, "", "Certificate authority file path to use to authenticate with Kubernetes.")
-	daemonCommand.PersistentFlags().String(f.Service.Kubernetes.TLS.CrtFile, "", "Certificate file path to use to authenticate with Kubernetes.")
-	daemonCommand.PersistentFlags().String(f.Service.Kubernetes.TLS.KeyFile, "", "Key file path to use to authenticate with Kubernetes.")
+	if err := mgr.AddHealthzCheck("healthz", healthz.Ping); err != nil {
```

Comment:
This may be an alternative to 7. [here](https://github.com/giantswarm/giantswarm/issues/17771#issuecomment-867471192).